L1Trigger/L1THGCal/src/backend_emulator/HGCalCluster_SA.cc

```diff
@@ -1,7 +1,245 @@
+  return intpart;
+}
+
+unsigned long int HGCalCluster::Sigma_eta_Fraction()  { 
+  double Sigma_eta_temp = Sigma_Coordinate(this->w(), this->weta2(), this->weta());
+  double intpart;
+  return modf(Sigma_eta_temp,&intpart)*pow(2,1);
+}
+
+unsigned long int HGCalCluster::Sigma_eta_Quotient()  { 
+  double Sigma_eta_temp = Sigma_Coordinate(this->w(), this->weta2(), this->weta());
+  double intpart;
+  double frac = modf(Sigma_eta_temp,&intpart)*pow(2,1);
+  return intpart;
+}
+
+unsigned long int HGCalCluster::Sigma_roz_Fraction()  { 
+  double Sigma_roz_temp = Sigma_Coordinate(this->w(), this->wroz2(), this->wroz());
+  double intpart;
+  return modf(Sigma_roz_temp,&intpart)*pow(2,1);
+}
+
+unsigned long int HGCalCluster::Sigma_roz_Quotient()  { 
+  double Sigma_roz_temp = Sigma_Coordinate(this->w(), this->wroz2(), this->wroz());
+  double intpart;
+  double frac = modf(Sigma_roz_temp,&intpart)*pow(2,1);
+  return intpart;
+}
+
+unsigned long int HGCalCluster::FirstLayer()  { 
+  std::vector<int> layeroutput = ShowerLengthProperties(this->layerbits());
+  return layeroutput[0];
+}
+
+unsigned long int HGCalCluster::LastLayer()  { 
+  std::vector<int> layeroutput = ShowerLengthProperties(this->layerbits());
+  return layeroutput[1];
+}
+
+unsigned long int HGCalCluster::ShowerLen()  { 
+  std::vector<int> layeroutput = ShowerLengthProperties(this->layerbits());
+  return layeroutput[2];
+}
+
+unsigned long int HGCalCluster::CoreShowerLen()  {
+  std::vector<int> layeroutput = ShowerLengthProperties(this->layerbits());
+  return layeroutput[3];
+}
+
+unsigned long int HGCalCluster::E_EM_over_E_Fraction()  { 
+  double E_EM_over_E_temp = Energy_ratio( this->e_em() , this->e() );
+  double intpart;
+  return modf(E_EM_over_E_temp,&intpart)*pow(2,8);
+}
+
+unsigned long int HGCalCluster::E_EM_over_E_Quotient()  { 
+  double E_EM_over_E_temp = Energy_ratio( this->e_em() , this->e() );
+  double intpart;
+  double frac = modf(E_EM_over_E_temp,&intpart)*pow(2,8);
+  return intpart;
+}
+
+unsigned long int HGCalCluster::E_EM_core_over_E_EM_Fraction()  { 
+  double E_EM_core_over_E_EM_temp = Energy_ratio( this->e_em_core() , this->e() );
+  double intpart;
+  return modf(E_EM_core_over_E_EM_temp,&intpart)*pow(2,8);
+}
+
+unsigned long int HGCalCluster::E_EM_core_over_E_EM_Quotient()  { 
+  double E_EM_core_over_E_EM_temp = Energy_ratio( this->e_em_core() , this->e() );
+  double intpart;
+  double frac =  modf(E_EM_core_over_E_EM_temp,&intpart)*pow(2,8);
+  return intpart;
+}
+
+unsigned long int HGCalCluster::E_H_early_over_E_Fraction()  { 
+  double E_H_early_over_E_temp = Energy_ratio( this->e_h_early() , this->e() );
+  double intpart;
+  return  modf(E_H_early_over_E_temp,&intpart)*pow(2,8);
+}
+
+unsigned long int HGCalCluster::E_H_early_over_E_Quotient()  { 
+  double E_H_early_over_E_temp = Energy_ratio( this->e_h_early() , this->e() );
+  double intpart;
+  double frac =  modf(E_H_early_over_E_temp,&intpart)*pow(2,8);
+  return intpart;
+}
+
+
```

Comment:
In general we may want to put the calculation of the various properties into a separate class, as is done in the existing CMSSW simulation, but I think we can leave this for now.

Reply:
done in 342389f